### PR TITLE
updates holochain validation so there are no deaths but cannot action…

### PR DESCRIPTION
**Pull request by willemolding**
… same position again

### CODE REVIEW

File: dna-src/minersweeper/index.ts [hunk 1]
```diff
@@ -129,15 +129,19 @@ function isDead(gameBoard: GameBoard, actions: Action[]): boolean {
 ============================================*/
 
 // the main validation function of the game. All game rules are enforced here
+// only allow actions to be taken on locations that have not aready been acted on
 function validateAddAction(gameHash, actionHash, agentHash) {
-  let gameBoard = get(gameHash);
+  let action: Action = get(actionHash);
+  if(action.actionType === "chat") return true;
```
Comment (maackle) [inline dna-src/minersweeper/index.ts]: note to us: maybe we should use some consts for the actionTypes now that we lost the enum...

File: ui-src/src/components/MiniMap.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+import './Cell.css';
+
+import {Hash} from '../../../holochain';
+
+import CellMatrix from '../CellMatrix';
+import {fetchJSON} from '../common'
+import store from '../store'
+
+interface IminiMapProps {
```
Comment (maackle) [inline ui-src/src/components/MiniMap.tsx]: I added a tslint rule that interfaces cannot start with "I" (the alternative is that they ALL must start with "I"), did you try "building" (yarn starting) this?

File: ui-src/src/components/MiniMap.tsx [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+      const buf8 = new Uint8ClampedArray(buf);
+      const data = new Uint32Array(buf);
+      for (let y = 0; y < canvas.height; ++y) {
+        for (let x = 0; x < canvas.width; ++x) {
+          const value = 4;
+            data[y * canvas.width + x] =
+              (255   << 24) |    // alpha
+              (value/2 << 16) |    // blue
+              (value <<  8) |    // green
+              255;            // red
```
Comment (maackle) [inline ui-src/src/components/MiniMap.tsx]: 😵

**Review by maackle — APPROVED** [review]
Wondering about one thing (did you try running this?) but looks good!

File: ui-src/src/components/MiniMap.tsx (hunk 2 above)
Comment (willemolding) [reply ui-src/src/components/MiniMap.tsx]: linter seems happy and I can build and run. I actually didn't mean to include this file though I guess I am getting sloppy. Was supposed to be back-end only changes.

File: ui-src/src/components/MiniMap.tsx (hunk 3 above)
Comment (willemolding) [reply ui-src/src/components/MiniMap.tsx]: I wish I could take credit for this

**Pull request merged**